### PR TITLE
Refactor acceptance tests

### PR DESCRIPTION
**Pull request by dcendents**
I know we wanted to run the acceptance tests on the oldest supported version as well as the latest. 

I think to be able to test functionalities where badges that used to be in the GroovyPostBuild plugin and in recent jenkins was moved to the badges plugin.

So in maven we have a module with the tests, a second module which runs the same tests but with a different value for `jenkins.version`. The parent project simply aggregates the serenity report. In eclipe we can only run the tests in the `base` project and not in `latest`.

I've refactor the tests to be parameterized instead and can now run against many jenkins versions, it is a single project now so it is easy to run the tests in eclipse as well.

### CODE REVIEW

**Comment by basil** [comment]
This looks great! BTW @dcendents it would be great if we could bump the minimum required Jenkins version to 2.289 or later and merge #472. That would allow the Jenkins core developers to remove a deprecated API that is only being used by this plugin.

**Comment by dcendents** [comment]
@basil to be fair I've been running a modified version of the build-monitor at work for many years and now I'm trying to fix the tests on master so I can rebase on it and make sure it all works. So to keep it simple I prefer this PR stays as is and only fix the acceptance tests and not modify any production code.

But I've rebased my PR #372 on this one and at the same time I bumped the version to 2.289.1 and replace all occurences of com.google.common.base.Predicate with the java 8 equivalent.

If that one can be merged I think we would both appreciate it.

File: build-monitor-acceptance/src/test/java/features/BuildMonitorAbstractBase.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@RunWith(SerenityParameterizedRunner.class)
+public abstract class BuildMonitorAbstractBase {
```
Comment (jan-molak) [inline build-monitor-acceptance/src/test/java/features/BuildMonitorAbstractBase.java:18]: I wonder if we could come up with a name other than `AbstractBase` that describes the purpose of this class a bit better. Maybe `BuilMonitorAcceptanceTest`?

**Comment by jan-molak** [comment]
Looks great, thanks @dcendents! I have just one minor suggestion, which I added to the review

**Review by jan-molak — CHANGES_REQUESTED** [review]

**Comment by jan-molak** [comment]
Merged, thanks! BTW: thank you for all your help, @dcendents and @basil, I really appreciate it :bow:

**Pull request merged**